### PR TITLE
[SYCL][CUDA] Allow FTZ, prec-sqrt to override no-ftz, no-prec-sqrt

### PR DESCRIPTION
**Pull request by hdelan**
If two bc files are compiled with different values for flags `nvvm-reflect-ftz` or `nvvm-reflect-prec-sqrt` then llvm-link will emit an error for the conflicting module flags. This instead allows FTZ=true to override FTZ=false, and the same with `prec-sqrt`. 

This was blocking ftz=true from being used in SYCL for CUDA backend as the `llvm/libdevice` library is compiled with default ftz value, meaning introducing a non default value for `nvvm-reflect-ftz` will fail at `llvm-link` time. 

An alternative is to introduce a clang flag that will completely omit nvvm-reflect module flags, which can then be used when compiling libdevice.

### CODE REVIEW

File: clang/lib/CodeGen/CodeGenModule.cpp [hunk 1]
```diff
@@ -850,12 +850,12 @@ void CodeGenModule::Release() {
     // Indicate whether __nvvm_reflect should be configured to flush denormal
     // floating point values to 0.  (This corresponds to its "__CUDA_FTZ"
     // property.)
-    getModule().addModuleFlag(llvm::Module::Override, "nvvm-reflect-ftz",
+    getModule().addModuleFlag(llvm::Module::Max, "nvvm-reflect-ftz",
```
Comment (premanandrao) [inline clang/lib/CodeGen/CodeGenModule.cpp:853]: Can you please explain what Max means here and Override meant before?

Comment (hdelan) [reply clang/lib/CodeGen/CodeGenModule.cpp:853]: See here https://github.com/intel/llvm/blob/sycl/llvm/include/llvm/IR/Module.h#L136 . Here Max means that the max value will become the global value at `llvm-link` time (which corresponds to FTZ=true). The `Override` option would instead give an error at llvm-link time for multiple definitions of the same module flag

Comment (premanandrao) [reply clang/lib/CodeGen/CodeGenModule.cpp:853]: Thank you!

**Comment by premanandrao** [comment]
Do we have tests in our suite to detect conflicts like this?

**Review by elizabethandrews — COMMENTED** [review]
Looks ok to me. I have the same question as @premanandrao. Is there a way to test that this change fixes the issue?

**Comment by hdelan** [comment]
Have added an ftz test here https://github.com/intel/llvm-test-suite/pull/1440

**Comment by hdelan** [comment]
/verify with https://github.com/intel/llvm-test-suite/pull/1440

**Review by premanandrao — APPROVED** [review]

**Review by smanna12 — APPROVED** [review]

**Comment by hdelan** [comment]
Can someone attach the logs from the llvm-test-suite failures please?

**Comment by dm-vodopyanov** [comment]
> Can someone attach the logs from the llvm-test-suite failures please?

```
[2022-12-07T17:54:30.745Z] Scheduling project: SYCL_CI » intel » Win » Test_Suite
[2022-12-07T17:54:39.417Z] Starting building: SYCL_CI » intel » Win » Test_Suite #2171
[2022-12-07T18:47:51.397Z] URL link: http://llvm-ci-test2.intel.com:8080//job/SYCL_CI/job/intel/job/Win/job/Test_Suite/2171/display/redirect
[2022-12-07T18:47:51.692Z] + exit -1
[2022-12-07T18:47:51.698Z] Err: Build failed with Error: hudson.AbortException: script returned exit code 255
[2022-12-07T18:47:51.977Z] + exit -1
script returned exit code 255
```

The entire logs, looks like some issue in Jenkins, the job could be re-started

**Comment by dm-vodopyanov** [comment]
/verify with https://github.com/intel/llvm-test-suite/pull/1440

**Review by elizabethandrews — APPROVED** [review]

**Comment by hdelan** [comment]
I don't think that the Jenkins build failures are related to this PR. Can this be merged?

**Comment by bader** [comment]
> I don't think that the Jenkins build failures are related to this PR. Can this be merged?

@hdelan, I'm sorry, it took so long. Let's update the test results. Could you sync your private branch with the tip of the branch, please?

**Comment by hdelan** [comment]
/verify with https://github.com/intel/llvm-test-suite/pull/1440

**Comment by againull** [comment]
/verify with https://github.com/intel/llvm-test-suite/pull/1440

**Comment by hdelan** [comment]
I'm not sure what's happening with Jenkins but I don't think it's related to this PR?

**Comment by againull** [comment]
L0 backend failures are because of the known problem ("/usr/bin/ld: cannot find -lze_loader"). This problem is fixed now.
HIP backend test failure is fixed.
Test failure in the Jenkins/llvm-test-suite  is known and tracked.

**Pull request merged**